docker-build/Dockerfile

```diff
@@ -0,0 +1,52 @@
+# Build stage
+FROM node:22-slim AS builder
+
+# Set working directory
+WORKDIR /app
+
+# Copy package files
+COPY package*.json ./
+
+# Install dependencies
+RUN npm ci
+
+# Copy source code
+COPY . .
+
+# Build the application
+RUN npm run build
+
+# Runtime stage
+FROM node:22-slim
+
+# Install tini
+RUN apt-get update && apt-get install -y tini curl && rm -rf /var/lib/apt/lists/*
+
+# Create non-root user
+RUN groupadd -r recoil && useradd -r -g recoil recoil
+
+# Set working directory
+WORKDIR /app
+
+# Copy built application from builder
+COPY --from=builder /app/dist ./dist
+COPY --from=builder /app/package*.json ./
+COPY --from=builder /app/node_modules ./node_modules
```

Comment:
You don't want to copy dev dependencies into production build. The production image should have only non-development dependencies.